cmd/validatorcmd/getBalance.go

```diff
@@ -58,68 +71,137 @@ func getBalance(_ *cobra.Command, _ []string) error {
 		return err
 	}
 
-	var balance uint64
-	if validationIDStr != "" {
-		validationID, err := ids.FromString(validationIDStr)
-		if err != nil {
-			return err
-		}
-		balance, err = txutils.GetValidatorPChainBalanceValidationID(network, validationID)
-		if err != nil {
-			return err
-		}
-		ux.Logger.PrintToUser("  Validator Balance: %.5f", float64(balance)/float64(units.Avax))
+	validationID, cancel, err := getNodeValidationID(network, l1, nodeIDStr, validationIDStr)
+	if err != nil {
+		return err
+	}
+	if cancel {
 		return nil
 	}
+	if validationID == ids.Empty {
+		return fmt.Errorf("the specified node is not a L1 validator")
+	}
 
-	isBootstrapValidator, err := app.Prompt.CaptureYesNo("Is the validator a bootstrap validator?")
+	balance, err := txutils.GetValidatorPChainBalanceValidationID(network, validationID)
 	if err != nil {
 		return err
 	}
-	if isBootstrapValidator {
-		if l1 == "" {
-			return fmt.Errorf("--l1 flag is required to get bootstrap validator balance")
-		}
-		sc, err := app.LoadSidecar(l1)
+	ux.Logger.PrintToUser("  Validator Balance: %.5f AVAX", float64(balance)/float64(units.Avax))
+
+	return nil
+}
+
+func getNodeValidationID(
+	network models.Network,
+	l1 string,
+	nodeIDStr string,
+	validationIDStr string,
+) (ids.ID, bool, error) {
+	var (
+		validationID ids.ID
+		err          error
+	)
+	if validationIDStr != "" {
+		validationID, err = ids.FromString(validationIDStr)
 		if err != nil {
-			return fmt.Errorf("failed to load sidecar: %w", err)
+			return validationID, false, err
 		}
-		if !sc.Sovereign {
-			return fmt.Errorf("avalanche validator getBalance command is only applicable to sovereign L1s")
+		return validationID, false, nil
+	}
+	l1ListOption := "I will choose from the L1 validators list"
+	validationIDOption := "I know the validation ID"
+	cancelOption := "Cancel"
+	option := l1ListOption
+	if l1 == "" {
+		options := []string{l1ListOption, validationIDOption, cancelOption}
+		option, err = app.Prompt.CaptureList(
+			"How do you want to specify the L1 validator",
+			options,
+		)
+		if err != nil {
+			return ids.Empty, false, err
 		}
-		bootstrapValidators := sc.Networks[network.Name()].BootstrapValidators
-		if len(bootstrapValidators) == 0 {
-			return fmt.Errorf("this L1 does not have any bootstrap validators")
+	}
+	switch option {
+	case l1ListOption:
+		chainSpec := contract.ChainSpec{
+			BlockchainName: l1,
 		}
-		bootstrapValidatorsString := []string{}
-		bootstrapValidatorsToIndexMap := make(map[string]int)
-		for index, validator := range bootstrapValidators {
-			bootstrapValidatorsString = append(bootstrapValidatorsString, validator.NodeID)
-			bootstrapValidatorsToIndexMap[validator.NodeID] = index
+		chainSpec.SetEnabled(true, false, false, false, true)
```

Comment:
this is very cryptic line. not sure what it means `true, false, false, false, true`

Reply:
looking at 
```
func (cs *ChainSpec) SetEnabled(
	blockchainNameFlagEnabled bool,
	cChainFlagEnabled bool,
	pChainFlagEnabled bool,
	xChainFlagEnabled bool,
	blockchainIDFlagEnabled bool,
) {
	cs.blockchainNameFlagEnabled = blockchainNameFlagEnabled
	cs.cChainFlagEnabled = cChainFlagEnabled
	cs.pChainFlagEnabled = pChainFlagEnabled
	cs.xChainFlagEnabled = xChainFlagEnabled
	cs.blockchainIDFlagEnabled = blockchainIDFlagEnabled
}
```
seems to be not that useful and more confusing when used.

Reply:
pls feel free to ignore this

Reply:
those 2 are not public

Reply:
adding a comment to improve legibility